### DISCUSSION TITLE
Preprocessing of scRNA reference

### DISCUSSION THREAD

**Discussion by livyring** [BayraktarLab/cell2location#188] [Q&A]
I am a bit confused on what level preprocessing is needed in order for the scRNA sequencing dataset to work correctly as a reference set in cell2location. My reference set was preprocessed and annotated using the standard scanpy tutorial workflow (identifying highly variable genes and clustering, etc). However, in the tutorial for the preprocessing of the scRNA reference in cell2location, you mention not to use variable genes, but rather a form of very permissive gene selection. However, I was under the assumption that the reference scRNA dataset must be annotated before loading it into cell2location (there is no annotation step in the tutorial, plus it looks like the reference set is already annotated in the obs.subset). 

I guess I'm just confused on what the more permissive gene selection method allows once the highly variable genes analysis has already been done upstream as a means for annotation. Should I be annotating my clusters/cell types apriori with the more permissive method rather than the high variable genes method? Or are they separable steps and both necessary in this analysis?

Thanks!!

**Comment by livyring** [comment]
![image](https://user-images.githubusercontent.com/106926355/182223818-c0c4d5bf-d335-4e35-a88c-5722160fa59f.png)
 when i go to plot my preprocessed data using:
 
from cell2location.utils.filtering import filter_genes
selected = filter_genes(adata_ref, cell_count_cutoff=5, cell_percentage_cutoff2=0.03, nonz_mean_cutoff=1.12)

I get the following graph. Can someone please help? Thank you!

**Reply by slee1779** [reply]
Did you find a solution to this?
